src/TagManager.lua

```diff
@@ -157,54 +164,75 @@ function TagManager:OnTagsUpdated(func)
 	return connection
 end
 
-function TagManager:_watchFolder()
-	for _, child in pairs(self.tagsFolder:GetChildren()) do
+function TagManager:_stopWatchingFolder(folder: Folder)
+	if self._maid[folder] then
+		self._maid[folder] = nil
+		self:_updateStore(true)
+	end
+end
+
+function TagManager:_watchFolder(folder: Folder)
+	if self._maid[folder]  then
+		return
+	end
+
+	local maid = Maid.new()
+
+	self._tagFolderSet[folder] = true
+	maid:give(function()
+		self._tagFolderSet[folder] = nil
+	end)
+
+	for _, child in pairs(folder:GetChildren()) do
 		if child:IsA("Configuration") then
 			self:_watchChild(child)
```

Comment:
```suggestion
			maid[instance] = self:_watchChild(child)
```
It looks like the maid gets thrown away here otherwise.